CDCgov/prime-reportstream
# 6934 - Change FHIR Path hasExtension function to hasPhoneNumberExtension

This PR refactors the `hasExtension` function used on phone numbers so that the name is clearer.

Test Steps:
1. Run test suite and verify everything passes.
2. Run the following command to convert a FHIR bundle to HL7 both in this branch and on master: `./prime fhirdata --input-file src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir -s metadata/hl7_mapping/ORU_R01/ORU_R01-base.yml --output-format=HL7`
3. Verify the output has not changed.

## Changes
- Renamed references to `hasExtension` and `HasExtension` on `ConstantResolver` and `PhoneUtilities` to include `PhoneNumber`.

## Linked Issues
- #6934
